### ISSUE TITLE
uninstall help

### ISSUE THREAD

**Issue by Kowy54** [gcgarner/IOTstack#197]
how to uninstall it?

**Comment by Slyke** [comment]
Hey @Kowy54 Uninstall what specifically? If you mean IOTstack, just run `docker-compse down` and then remove the IOTstack directory. You may also want to prune all images. Home Assistant has its own uninstall instructions, if you had installed that, we just wrap their installer.

**Comment by Kowy54** [comment]
I cant get rid off all of installed containers by IOTstack :/

**Comment by Paraphraser** [comment]
Start by doing a `docker images`

```
$ docker images
REPOSITORY            TAG                 IMAGE ID            CREATED             SIZE
kunde21/gitea-arm     latest              b1320f20c065        7 days ago          104MB
grafana/grafana       latest              ba22a0a392bb        12 days ago         149MB
portainer/portainer   latest              dbf28ba50432        2 weeks ago         62.5MB
influxdb              latest              f9c2ff6c2a14        2 weeks ago         261MB
httpd                 latest              8242588f7ded        2 weeks ago         126MB
pihole/pihole         latest              24f577969730        3 weeks ago         301MB
iotstack_nodered      latest              dd4a3bb2faf9        3 weeks ago         442MB
nodered/node-red      latest              b7b447a3e061        3 weeks ago         377MB
eclipse-mosquitto     latest              a2b55301913b        2 months ago        5.86MB
```

Then, for each value in the "IMAGE ID" column, try to remove the image:

```
$ docker rmi b1320f20c065
```

It will either work or say that it can't do that because a stopped container is using it. If you see such a message, it will end with a container ID. What you do then is:

```
$ docker rm <container ID from error message>
```

Then, retry the `rmi` of the image ID. Keep iterating like that and, eventually, all the stopped containers and images will be gone. There may be faster ways of doing this but that's what I do.

**Comment by Kowy54** [comment]
If i delete it, they show up again and again :/

**Comment by Paraphraser** [comment]
If they are coming back then my guess is that you might not have stopped the stack with

```
$ cd ~/IOTstack
$ docker-compose down 
```

A bit of Googling found [this hint](https://mytrashcode.com/stop-remove-all-docker-containers). I have not tried those commands but I see no reason why they would not work.

**Comment by Kowy54** [comment]
![2020-08-06 18_34_23-Window](https://user-images.githubusercontent.com/31593455/89557695-81b01400-d813-11ea-9d58-bcd0312fa04a.png)

**Comment by Paraphraser** [comment]
I really do not know what else to suggest. I think the best advice I can give is to start over from a clean install of Raspbian and faithfully observe the two basic assumptions of IOTstack:

1. The user **must** be the standard default user (username "pi", user ID 1000).
2. The install directory **must** be ~/IOTstack (which must expand to /home/pi/IOTstack as that exact spelling).

Those two assumptions were inherited from gcgarner/IOTstack. I believe Slyke is working to remove these dependencies in this fork but I do not know how far along he is nor whether his work has made it out of the experimental stage.

I am not saying that this is the true cause of what you are seeing. I'm simply observing `/home/kowy` in those error messages and interpreting it as evidence of "creativity", by which I mean departing from those two assumptions, at least, and possibly other roll-your-own decisions besides. Am I wrong to interpret the error messages like that?

I am not criticising either your decisions or your reasons. It's just that all my experience with IOTstack (about 10 months) both hands-on and watching issues come and go is that "IOTstack" and "creative" do not really belong in the same sentence. Those two assumptions are embedded quite deeply so it's better not to fight them.

**Comment by Kowy54** [comment]
I´m not running on pi but NUC with Debian 10

**Comment by Paraphraser** [comment]
I don't think that changes my advice. A lot of things rely on those assumptions and it would not surprise me if departing from those assumptions is part of the basic problem.

That said:

* If you had mentioned "NUC with Debian 10" before, I would not have tried to help - for the very simple reason that I really have no experience with that platform (probably obvious to you because I failed to pick up on the "@NUC" in your output above). Please accept my apologies for any unintended confusion I may have introduced into the process.
* the opening description on this site is "IOTstack is a builder for docker-compose to easily make and maintain IoT stacks **on the Raspberry Pi**" (my emphasis). Hopefully, someone else monitoring this (mostly defunct) project who understands your runtime environment will be able to assist.

> By "mostly defunct" I mean that this gcgarner/IOTstack project has been forked to [SensorsIot/IOTstack](https://github.com/SensorsIot/IOTstack) because Graham Garner has not been heard from for quite some time. We all hope he is OK but nobody seems to know.

**Comment by nuc** [comment]
Can confirm, it's not running on me.
